contributor_docs/hi/contributor_guidelines.md

```diff
@@ -0,0 +1,530 @@
+
+टर्मिनल से, जब आप `मुख्य` शाखा पर हों तो `git checkout -b ब्रांच_नाम` चलाएँ, `branch_name` को किसी वर्णनात्मक चीज़ से बदलें, और अब आप एक अलग शाखा पर होंगे।
+
+जैसे ही आप अपने परिवर्तन करते हैं, हम `npm test` को बार-बार चलाने की सलाह देते हैं, खासकर यदि आप स्रोत कोड पर काम कर रहे हैं। इसे चलाने में कुछ समय लगेगा, लेकिन यह सुनिश्चित करता है कि आपके द्वारा किए गए परिवर्तन मौजूदा व्यवहार को नहीं तोड़ रहे हैं। नीचे बताए अनुसार परिवर्तन करने से पहले आपको `npm test` चलाना चाहिए।
+
+एक बार जब आप कोडबेस में अपने परिवर्तन कर लेते हैं, तो आपको इसे गिट में प्रतिबद्ध करना होगा। कमिट गिट रिपॉजिटरी में सहेजे गए परिवर्तनों का एक संग्रह है; यह अनिवार्य रूप से प्रतिबद्धता के समय रेपो में फ़ाइलों की वर्तमान स्थिति को रिकॉर्ड करता है।
+
+एक प्रश्न उठ सकता है कि आपको कितनी बार गिट के लिए प्रतिबद्ध होना चाहिए? सामान्य तौर पर यह प्राथमिकता दी जाती है कि आप कई बड़े बदलावों को एक प्रतिबद्धता में डालने के बजाय बार-बार प्रतिबद्ध होने का लक्ष्य रखें। एक अच्छा दिशानिर्देश यह है कि जब भी आप कोई ऐसा उपकार्य पूरा कर लें जिसे एक वाक्य में वर्णित किया जा सके तो उसे प्रतिबद्ध कर दें।
+
+गिटहब डेस्कटॉप से ​​सभी मौजूदा परिवर्तन करने के लिए, अपने परिवर्तन करने के बाद ऐप खोलें। यह बाएं साइडबार में आपके द्वारा बदली गई फ़ाइलों की एक सूची दिखाएगा, और दाईं ओर प्रत्येक फ़ाइल के भीतर विशिष्ट परिवर्तन दिखाएगा। विंडो के निचले बाएँ कोने में अपने उपयोगकर्ता आइकन के त्रुटिल वाले फ़ील्ड में एक संक्षिप्त, उच्च-स्तरीय विवरण टाइप करें। यह कमिट का शीर्षक होगा. आप नीचे विवरण फ़ील्ड में और विस्तार से बता सकते हैं या इसे खाली छोड़ सकते हैं। परिवर्तन को अंतिम रूप देने के लिए नीले "प्रतिबद्ध" बटन पर क्लिक करें।
+
+![परिवर्तन करने के बाद गिटहब डेस्कटॉप का एक स्क्रीनशॉट। वह क्षेत्र जहां आपको अपने परिवर्तन के लिए शीर्षक लिखना है, विंडो के नीचे बाईं ओर लाल रंग में दर्शाया गया है।](../images/github-desktop-commit.png)
+
+टर्मिनल से सभी मौजूदा परिवर्तन करने के लिए, निम्नलिखित चलाएँ:
+
+1. जांचें कि यह केवल उन फ़ाइलों को सूचीबद्ध करता है जिन्हें आपने निम्न आदेश के साथ बदला है।
+
+```
+git status
+```
+
+यदि सूचीबद्ध फ़ाइलें हैं जिन्हें आपने नहीं बदला है, तो आपको या तो उन्हें मूल रूप में [पुनर्स्थापित](https://git-scm.com/docs/git-restore) करना होगा या सुनिश्चित करना होगा कि वे इच्छित परिवर्तन हैं। प्रत्येक फ़ाइल के लिए अधिक विस्तृत परिवर्तन दिखाने के लिए निम्न आदेश का उपयोग करें।
+
+```
+git diff
+```
+
+आपको कोई भी फ़ाइल परिवर्तन नहीं करना चाहिए जिसे आप अपने पीआर के लिए बदलने का इरादा नहीं रखते हैं।
+
+2. निम्नलिखित आदेश के साथ गिट में प्रतिबद्ध होने के लिए सभी परिवर्तनों को चरणबद्ध करें।
+
+```
+git add .
+```
+
+3. गिट में परिवर्तन करने के लिए, निम्न कमांड चलाएँ।
+
+```
+git commit -m "[your_commit_message]"
+```
+
+`[your_commit_message]` को एक प्रासंगिक प्रतिबद्ध संदेश के साथ प्रतिस्थापित किया जाना चाहिए जो सामान्य बयानों से बचते हुए परिवर्तनों का वर्णनात्मक हो। उदाहरण के लिए, 'डॉक्यूमेंटेशन फिक्स 1' कहने के बजाय, 'सर्कल() फ़ंक्शन में दस्तावेज़ उदाहरण जोड़ें' कहें।
```

Comment:
can we make this text a bit more simplier?

Reply:
Done